test/demo/recreate.inttest.ts

```diff
@@ -0,0 +1,157 @@
+                await delay(200);
+            }
+        });
+
+        afterAll(async () => {
+            // Close center when we're done
+            if (await isCenterShowing()) {
+                await notifsRemote.toggleNotificationCenter(testManagerIdentity);
+                await delay(Duration.centerToggled);
+            }
+        });
+
+        beforeEach(async () => {
+            testApp = await createApp(testManagerIdentity, {});
+            testWindow = await testApp.getWindow();
+
+            // Quick sanity check that there is not already a notification with that ID
+            const notes = await notifsRemote.getAll(testWindow.identity);
+            expect(notes.some(note => note.id === firstOptions.id)).not.toBeTruthy();
+
+            // Create the "exisitng" notificaiton
+            existingNote = await notifsRemote.create(testWindow.identity, firstOptions);
+        });
+
+        afterEach(async () => {
+            await notifsRemote.clearAll(testWindow.identity);
+            await testApp.quit(true).catch(() => {});
+        });
+
+        test('The promise resolves to the new notification object', async () => {
+            const newNotePromise = notifsRemote.create(testWindow.identity, secondOptions);
+            expect(newNotePromise).resolves;
+
+            expect(await newNotePromise).toMatchObject(secondOptions);
+        });
+
+        test('No "cleared" event is emitted', async () => {
+            // Register a closed listener on the window
+            const closeListener = jest.fn<void, [NotificationClosedEvent]>();
+            await notifsRemote.addEventListener(testWindow.identity, 'notification-closed', closeListener);
+
+            // Recreate the notification and pause momentarily to allow the event to propagate
+            await notifsRemote.create(testWindow.identity, secondOptions);
+            await delay(Duration.eventPropagated);
+
+            // Listener not triggered
+            expect(closeListener).not.toHaveBeenCalled();
+        });
+
+        test('The card in the center is updated to show the new notification details', async () => {
+            // Existing card matches `firstOptions`
+            await assertDOMMatches(CardType.CENTER, testApp.identity.uuid, existingNote);
+
+            // Recreate the notification
+            const newNote = await notifsRemote.create(testWindow.identity, secondOptions);
+
+            // New card matches `secondOptions`
+            await assertDOMMatches(CardType.CENTER, testApp.identity.uuid, newNote);
+        });
+
+        // Extra tests for toasts only when the center is hidden
+        if (!showCenter) {
+            describe('When the existing notification has an active toast', () => {
+                const finOpenListener = jest.fn<void, [WindowEvent<'system', 'window-created'>]>();
+                const finCloseListener = jest.fn<void, [WindowEvent<'system', 'window-closed'>]>();
```

Comment:
The same callbacks will be used for all tests in this block - might result in state leaking between tests?..

Reply:
Each test is run in it's own execution context. Anything in memory in the actual test run is not persisted between tests (with the exception of things defined in the top level of the jest environment file). 

At least that's how I've understood it. If that's not the case can you point me to the docs or some code that demonstrates how it actually works.

Reply:
Each test _file_ is ran in it's own execution context, but if you put a `console.log` here in this scope, I think you'd only see it the once.

This is based more on my understanding of how mocha works, than Jest though. Jest _could_ possibly be parsing the test file for every single test, though I'm assuming anything in this kind of scope only runs the once, before any tests start.

Reply:
Fixed as discussed